### PR TITLE
docs: add TypeScript code blocks in ISR page

### PR DESCRIPTION
**Pull request by Juneezee**
Currently only the code block in the first section of the ISR page has both TypeScript and JavaScript code. This PR updates the remaining code blocks to include TypeScript code.

/cc original author of ISR page @leerob

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: b9b0c01312b6a8e1e855a42059780546b5fb6c88

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: b9b0c01312b6a8e1e855a42059780546b5fb6c88

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: b9b0c01312b6a8e1e855a42059780546b5fb6c88

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx [hunk 1]
```diff
@@ -228,7 +232,32 @@ Here's how this example works:
...
+        {posts.map((post) => (
+          <li key={post.id}>{post.title}</li>
+        ))}
+      </ul>
+    </main>
+  )
+}
+```
+
+```jsx filename="app/blog/page.jsx" switcher
```
Comment (leerob) [inline docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx:260]:
```suggestion
```jsx filename="app/blog/page.js" switcher
```

File: docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx [hunk 2]
```diff
@@ -64,28 +64,32 @@ export default async function Page({ params }: { params: { id: string } }) {
 ```jsx filename="app/blog/[id]/page.jsx" switcher
```
Comment (leerob) [inline docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx:64]:
```suggestion
```jsx filename="app/blog/[id]/page.js" switcher
```

File: docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx [hunk 3]
```diff
@@ -272,7 +312,17 @@ export async function createPost() {
...
+export default async function Page() {
+  const data = await fetch('https://api.vercel.app/blog', {
+    next: { tags: ['posts'] },
+  })
+  const posts = await data.json()
+  // ...
+}
+```
+
+```jsx filename="app/blog/page.jsx" switcher
```
Comment (leerob) [inline docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx:325]:
```suggestion
```jsx filename="app/blog/page.js" switcher
```

File: docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx [hunk 4]
```diff
@@ -284,7 +334,25 @@ export default async function Page() {
...
+  { revalidate: 3600, tags: ['posts'] }
+)
+
+export default async function Page() {
+  const posts = getCachedPosts()
+  // ...
+}
+```
+
+```jsx filename="app/blog/page.tsx" switcher
```
Comment (leerob) [inline docs/02-app/01-building-your-application/02-data-fetching/04-incremental-static-regeneration.mdx:355]:
```suggestion
```jsx filename="app/blog/page.js" switcher
```

**Review by leerob — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**